From e58d6dd9b5c7a9a6fb8c7d3b9c443bd3d354ae40 Mon Sep 17 00:00:00 2001
From: James Turton <james.turton@gmx.com>
Date: Tue, 10 Mar 2026 00:14:08 +0100
Subject: [PATCH] Don't wait for mcu to respond after flashing

---
 mcu-flasher/src/ymodem.rs | 26 --------------------------
 1 file changed, 26 deletions(-)

diff --git a/mcu-flasher/src/ymodem.rs b/mcu-flasher/src/ymodem.rs
index c894a49..9e52631 100644
--- a/mcu-flasher/src/ymodem.rs
+++ b/mcu-flasher/src/ymodem.rs
@@ -402,32 +402,6 @@ impl Ymodem {
             }
         }
 
-        loop {
-            match (get_byte_timeout(dev))? {
-                Some(c) => {
-                    if c == CRC {
-                        info!("YMODEM transmission successful");
-                        break;
-                    } else {
-                        log::warn!("Expected ACK, got {}", c);
-                    }
-                }
-                None => warn!("Timeout waiting for ACK for EOT"),
-            }
-
-            self.errors += 1;
-
-            if self.errors >= self.max_errors {
-                eprint!(
-                    "Exhausted max retries ({}) while waiting for ACK for EOT",
-                    self.max_errors
-                );
-                return Err(Error::ExhaustedRetries);
-            }
-        }
-
-        self.send_end_frame(dev)?;
-
         Ok(())
     }
 
