From 1ccfa5b77ba96b9e0ba9ed5ab4770db9f6cf2ecf Mon Sep 17 00:00:00 2001
From: James Turton <james.turton@gmx.com>
Date: Mon, 6 Apr 2026 23:18:37 +0200
Subject: [PATCH] Webcam allow ports in local url

---
 moonraker/components/webcam.py | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/moonraker/components/webcam.py b/moonraker/components/webcam.py
index 6c33b88..1227a64 100644
--- a/moonraker/components/webcam.py
+++ b/moonraker/components/webcam.py
@@ -261,8 +261,19 @@ class WebCam:
                 f"Invalid value for 'aspect_ratio': {self.aspect_ratio}"
             )
 
+    def _expand_url(self, url: str) -> str:
+        if re.match(r"^:\d+/", url):
+            return f"{self._default_host}{url}"
+        match = re.match(r"\w+://[^/]+", url)
+        if match is None:
+            return f"{self._default_host}/{url.lstrip('/')}"
+        return url
+
     def as_dict(self):
-        return {k: v for k, v in self.__dict__.items() if k[0] != "_"}
+        d = {k: v for k, v in self.__dict__.items() if k[0] != "_"}
+        d["stream_url"] = self._expand_url(self.stream_url)
+        d["snapshot_url"] = self._expand_url(self.snapshot_url)
+        return d
 
     async def get_stream_url(self, convert_local: bool = False) -> str:
         return await self._get_url(self.stream_url, convert_local)
@@ -273,10 +284,7 @@ class WebCam:
     async def _get_url(self, url: str, convert_local: bool) -> str:
         if not url:
             raise self._server.error("Empty URL Provided")
-        match = re.match(r"\w+://[^/]+", url)
-        if match is None:
-            # assume a partial URL on the default host
-            url = f"{self._default_host}/{url.lstrip('/')}"
+        url = self._expand_url(url)
         if not convert_local:
             return url
         return await self.convert_local(url)
